src/server/rdb_save.cc

```diff
@@ -1201,28 +1203,58 @@ error_code RdbSaver::Impl::ConsumeChannel(const Cancellation* cll) {
         continue;
 
       DVLOG(2) << "Pulled " << record.id;
-      last_write_time_ns_ = absl::GetCurrentTimeNanos();
-      io_error = sink_->Write(io::Buffer(record.value));
+      auto start = absl::GetCurrentTimeNanos();
+      io_error = WriteRecord(io::Buffer(record.value));
+      if (io_error) {
+        break;  // from the inner TryPop loop.
+      }
 
-      stats.rdb_save_usec += (absl::GetCurrentTimeNanos() - last_write_time_ns_) / 1'000;
+      auto delta_usec = (absl::GetCurrentTimeNanos() - start) / 1'000;
+      stats.rdb_save_usec += delta_usec;
       stats.rdb_save_count++;
       last_write_time_ns_ = -1;
```

Comment:
I think we can move last_write_time_ns_ = -1 inside WriteRecord after we finish the loop

Reply:
done